src/main/java/teammates/storage/api/InstructorsDb.java

```diff
@@ -502,12 +503,35 @@ private Instructor getInstructorEntityForRegistrationKey(String key) {
         q.declareParameters("String googleIdParam");
         q.setFilter("googleId == googleIdParam");
         
+        
         @SuppressWarnings("unchecked")
         List<Instructor> instructorList = (List<Instructor>) q.execute(googleId);
         
         return instructorList;
     }
     
+    /**
+     * Omits instructors with isArchived == omitArchived.
+     * This means that the corresponding course is archived by the instructor.
+     */
+    private List<Instructor> getInstructorEntitiesForGoogleId(String googleId, boolean omitArchived) {
+        
+        if (omitArchived) {
+            Query q = getPM().newQuery(Instructor.class);
+            q.declareParameters("String googleIdParam, boolean omitArchivedParam");
+            // Omit archived == true, get instructors with isArchived != true
+            q.setFilter("googleId == googleIdParam && isArchived != omitArchivedParam");
+            
+            @SuppressWarnings("unchecked")
+            List<Instructor> instructorList = (List<Instructor>) q.execute(googleId, omitArchived);
+            System.out.println("1- " + instructorList.size());
+            System.out.println("2- " + getInstructorEntitiesForGoogleId(googleId).size());
+            return instructorList;
+        } else {
+            return getInstructorEntitiesForGoogleId(googleId);
```

Comment:
Consider writing the shorter branch first. Readability is better when the `if` and `else` are closer.